bin/faucet/src/main.rs

```diff
@@ -72,6 +73,26 @@ async fn main() -> Result<(), FaucetError> {
 
             info!(target: COMPONENT, %config, "Initializing server");
 
+            let client_state_clone = Arc::clone(&faucet_state.client);
+
+            info!("Initializing chain tip updater");
+            thread::spawn(move || {
+                let sys = actix_web::rt::System::new();
+                sys.block_on(async move {
+                    loop {
+                        let state_clone_inner = Arc::clone(&client_state_clone);
+                        actix_web::rt::spawn(async move {
+                            let mut state = state_clone_inner.lock().await;
+                            state.update_current_block_number().await.unwrap();
```

Comment:
I don't have a strong opinion here - but I'm also thinking that maybe we could implement a solution mentioned by @igamigo in https://github.com/0xPolygonMiden/miden-node/issues/440#issuecomment-2297068005. What are the downsides of doing this?

Reply:
I think a downside is that the block number used is always one transaction behind I think? So request `N` will use the block number as returned by transaction `N-1` unless I've misunderstood the order of operations?

This makes the block number staleness unbounded as it depends on transaction frequency. With a separate task its bounded by the poll period.

Reply:
I think which block number we use for the transaction is less important here than the block number we attach to the generated note (this is the [after_block_num](https://github.com/0xPolygonMiden/miden-base/blob/next/objects/src/notes/file.rs#L23) hint which the client can then use to limit its search for the note). And since we can attach this after the note is created, I think the sequence of operations could be:

1. Create a transaction using the latest known `block_num` (e.g., $N$).
2. Submit the transaction to the node and get back the new latest `block_num` (e.g., $M$).
3. Set `after_block_num` for the note to $M$ - this way, the user gets the latest possible block number.
4. Save $M$ as the new latest `block_num`.

Yes, the next transaction would use $M$ as `block_num` and it may be "out of date" by that time, but what really matters here (I think) is step 3 above.

Reply:
Thanks that makes sense. 

ito implementation for the faucet this would be fairly straight-forward I think. The handler already holds a lock for the duration of the query/execution so that part should be simple.

iiuc the only other change would be adding this to the proto response.

@SantiagoPittella how do you feel about trying method this instead?

Reply:
I've been working on adding the `chain_tip` to the response of `submit_proven_transaction`. To do this, I needed to add an endpoint to the Store: `get_chain_tip`. Since I don't have access to `State` or `Store` from the `block-producer` component, the only way I see to make use of this new endpoint is by adding a call in [`crates/rpc/src/server/api.rs:190`](https://github.com/0xPolygonMiden/miden-node/blob/next/crates/rpc/src/server/api.rs#L190), which I understood to be just a dispatcher to the corresponding component, and modifying the `block-producer`'s response with the correct chain_tip. Something like:
```rust
        // Fetch the store for the chain_tip.
        let chain_tip = self.store.clone().get_chain_tip(GetChainTipRequest {} ).await?.into_inner().chain_tip;

        // Call the block-producer's submit_proven_transaction
        let mut response = self.block_producer.clone().submit_proven_transaction(request).await?;

        // Modify response adding the chain_tip
        response.chain_tip = chain_tip;
```
This doesn't seem like a good idea, but I don't see any other way to do it. Any help or workaround would be greatly appreciated.

Reply:
I think we may be able to avoid adding a new endpoint. For every transaction, the block producer makes a `GetTransactionInputs` request to the store. I think we may be able to include the chain tip in the [GetTransactionInputsResponse](https://github.com/0xPolygonMiden/miden-node/blob/next/proto/responses.proto#L126). And then, the block producer would return it to the RPC component.

Reply:
Though, if it does turn out that passing this info through creates too many complications, we can switch to the original approach.

Reply:
I removed the endpoint and used the new approach. I think the code quality is much better with version. Let me know what you think.